private/tests/cross_platform/BUILD.bazel

```diff
@@ -0,0 +1,143 @@
+load("@rules_cc//cc:defs.bzl", "cc_toolchain")
+load("@rules_python//python:defs.bzl", "py_binary", "py_test")
+load("//:defs.bzl", "py_reqs_compiler", "py_reqs_solution_test")
+load(":cross_platform_test_utils.bzl", "fake_cc_config", "platform_transitioned_output_group")
+
+py_reqs_compiler(
+    name = "requirements.linux.update",
+    requirements_in = "requirements.in",
+    requirements_txt = "requirements.linux.txt",
+    target_compatible_with = ["@platforms//os:linux"],
+)
+
+py_reqs_solution_test(
+    name = "requirements_linux_test",
+    compiler = ":requirements.linux.update",
+)
+
+py_reqs_compiler(
+    name = "requirements.macos.update",
+    requirements_in = "requirements.in",
+    requirements_txt = "requirements.macos.txt",
+    target_compatible_with = ["@platforms//os:macos"],
+)
+
+py_reqs_solution_test(
+    name = "requirements_macos_test",
+    compiler = ":requirements.macos.update",
+)
+
+py_reqs_compiler(
+    name = "requirements.windows.update",
+    requirements_in = "requirements.in",
+    requirements_txt = "requirements.windows.txt",
+    target_compatible_with = ["@platforms//os:windows"],
+)
+
+py_reqs_solution_test(
+    name = "requirements_windows_test",
+    compiler = ":requirements.windows.update",
+)
+
+alias(
+    name = "requirements.update",
+    actual = select({
+        "@platforms//os:linux": "requirements.linux.update",
+        "@platforms//os:macos": "requirements.macos.update",
```

Comment:
On the bzlmod branch tests are kind of going to work on intel macs.  I can switch this to use my //private/settings:macos_aarch64 setting instead of just @platforms//os:macos

Reply:
You should definitely use aarch64 macs as they're what apple is investing in. All the macos Bazel things in the repo are all aarch64. As long as you're using the `macos-14` runner in your actions you'll also be using aarch64 in your Github Actions too.

Reply:
Thanks I'll just pop in a new CPU